reggie/ingestion/preprocessor/washington_preprocessor.py

```diff
@@ -45,18 +50,34 @@ def execute(self):
         voter_file = [n for n in new_files if "vrdb" in n["name"].lower()][0]
         hist_files = [n for n in new_files if "history" in n["name"].lower()]
 
+        # There are two possible separators. Detect it first.
+        line = voter_file["obj"].readline().decode()
+        delimiter = detect(line)
+        # Return to the beginning of the buffer to read the data now that we
+        # know what the separator is.
+        voter_file["obj"].seek(0)
         df_voter = pd.read_csv(
-            voter_file["obj"], sep="\t", encoding="latin-1", dtype=str
-        )
-        df_hist = pd.concat(
-            [
-                pd.read_csv(n["obj"], sep="\t", encoding="latin-1", dtype=str)
-                for n in hist_files
-            ],
-            ignore_index=True,
+            voter_file["obj"], sep=delimiter, encoding="latin-1", dtype=str
         )
+        df_hist = pd.DataFrame()
+        for hist_file in hist_files:
+            line = hist_file["obj"].readline().decode()
+            delimiter = detect(line)
+            hist_file["obj"].seek(0)
+            temp = pd.read_csv(
+                hist_file["obj"], sep=delimiter, encoding="latin-1", dtype=str
+            )
+            df_hist = df_hist.append(temp, ignore_index=True)
 
         # --- handling the voter history file --- #
+        # Need to fix/combine the differently named VoterHistoryID
+        # and VotingHistoryID columns
+        if {"VotingHistoryID", "VoterHistoryID"}.issubset(df_hist.columns):
+            df_hist["VotingHistoryID"] = (
+                df_hist.pop("VoterHistoryID").fillna(
+                    df_hist.pop("VotingHistoryID")
+                )
+            )
 
         # can't find voter history documentation in any yaml, hardcoding column name
         election_dates = pd.to_datetime(
```

Comment:
I wasn't sure what to make of this part. I _think_ this should probably go in the config?

Reply:
^ @cvcotton

Reply:
I think this is just coercing the dates in the history file (which might have a different format from the main voter file, so maybe she couldn't use the "coerce_dates" method?)